### PR TITLE
Add proper return code for i3-msg

### PR DESCRIPTION
**Pull request by NilsIrl**
I thought I might do a PR even though it had already been discussed at #1639.

Also maybe making the return code 1 for this situation isn't the best thing maybe -1 would be preferable since it would collide with EXIT_FAILURE.

### CODE REVIEW

**Comment by Airblader** [comment]
I think this is worthwhile, because sometimes just knowing a command failed is all you need to know. Given that @stapelberg explicitly rejected this in the past I'd like him to reconsider here, though.

**Comment by stapelberg** [comment]
I’m not entirely opposed to the change (it’s simple enough and returning a non-0 error code when things went wrong is common behavior), but the rationale provided in #1639 still stands: most cases should really not be using i3-msg to begin with.

Can you outline what your use-case is please?

**Comment by NilsIrl** [comment]
It is exactly was is described in #1639:
```bash
i3-msg [instance="$1"] scratchpad show || $2
```
currently what I have to do instead is this:
```bash
if [ $(i3-msg "[instance=\"^$1\$\"] scratchpad show") != '[{"success":true}]' ]; then
    $2
fi
```
Though they could be other use cases.

File: i3-msg/main.c [hunk 1]
```diff
@@ -100,6 +101,7 @@ static int reply_end_map_cb(void *params) {
             fprintf(stderr, "ERROR:               %s\n", last_reply.errorposition);
         }
         fprintf(stderr, "ERROR: %s\n", last_reply.error);
+        exit_code = 1;
```
Comment (stapelberg) [inline i3-msg/main.c]: Let’s make this exit code 2 and document the exit code in the man page.

File: man/i3-msg.man [hunk 2]
```diff
@@ -90,6 +90,15 @@ See the -m option for continuous monitoring.
...
 interface to i3.
 
+=== Exit status:
+
+0:
+if OK,
+1:
+if invalid syntax or unable to connect to ipc-socket
+2:
+if error while executing message
```
Comment (stapelberg) [inline man/i3-msg.man]: for clarity, how about: i3 returned an error processing your command(s)

**Comment by Airblader** [comment]
@NilsIrl Sorry for the wait; if you drop a comment after making the changes we get notified and can take another look. I've reviewed it now, though. Thanks for the PR!

**Comment by jwflory** [comment]
Hey, did this PR resolve #3614 too by chance?

**Comment by Airblader** [comment]
I can't take a look right now but I would suspect that it doesn't. :-)

**Pull request merged**